packages/core/src/lib/keep_alive_manager.ts

```diff
@@ -33,9 +67,15 @@ export class KeepAliveManager {
 
     if (pingPeriodSecs !== 0) {
       const interval = setInterval(() => {
-        libp2pPing.ping(peerId).catch((e) => {
-          log(`Ping failed (${peerIdStr})`, e);
-        });
+        this.libp2pPing
+          .ping(peerId)
+          .then((ping) => {
+            log(`Ping succeeded (${peerIdStr})`, ping);
+            this.peerPings.set(peerIdStr, ping);
```

Comment:
I think we should strive to re-use architecture of `libp2p` and change design here a bit. 

What I want to propose is to use `PeerStore` to store pings.

```ts
this.libp2pPing
  .ping(peerId)
  .then((ping) => {
    this.libp2p.peerStore.merge(peerId, {
      metadata: {
        ping,
      }
    });
  });
```

this way we:
- keep the responsibility of `KeepAliveManager` to only one thing;
- `PeerStore` keeps it's responsibility to store stuff about peer;
- other bonus points such as ability to subscribe, no need to change `selectPeerForProtocol` much, solution is integrated into `libp2p`;